server/src/main/java/io/unitycatalog/server/persist/ModelRepository.java

```diff
@@ -0,0 +1,361 @@
+                  session, catalogName, schemaName, updateRegisteredModel.getNewName());
+          if (newRegisteredModelInfoDAO != null) {
+            throw new BaseException(
+                ErrorCode.ALREADY_EXISTS, "Registered model already exists: " + newFullName);
+          }
+        }
+        // Get the record from the database
+        RegisteredModelInfoDAO origRegisteredModelInfoDAO =
+            findRegisteredModel(session, catalogName, schemaName, registeredModelName);
+        if (origRegisteredModelInfoDAO == null) {
+          throw new BaseException(ErrorCode.NOT_FOUND, "Registered model not found: " + fullName);
+        }
+        if (updateRegisteredModel.getNewName() != null) {
+          origRegisteredModelInfoDAO.setName(updateRegisteredModel.getNewName());
+        }
+        if (updateRegisteredModel.getComment() != null) {
+          origRegisteredModelInfoDAO.setComment(updateRegisteredModel.getComment());
+        }
+        long updatedTime = System.currentTimeMillis();
+        origRegisteredModelInfoDAO.setUpdatedAt(new Date(updatedTime));
+        session.persist(origRegisteredModelInfoDAO);
+        registeredModelInfo = origRegisteredModelInfoDAO.toRegisteredModelInfo();
+        registeredModelInfo.setCatalogName(catalogName);
+        registeredModelInfo.setSchemaName(schemaName);
+        registeredModelInfo.setFullName(getRegisteredModelFullName(registeredModelInfo));
+        tx.commit();
+      } catch (RuntimeException e) {
+        if (tx != null && tx.getStatus().canRollback()) {
+          tx.rollback();
+        }
+        throw e;
+      }
+    } catch (RuntimeException e) {
+      if (e instanceof BaseException) {
+        throw e;
+      }
+      throw new BaseException(
+          ErrorCode.INTERNAL, "Error updating registered model: " + fullName, e);
+    }
+    return registeredModelInfo;
+  }
+
+  public void deleteRegisteredModel(String fullName, boolean force) {
+    LOGGER.info("Deleting Registered Model: " + fullName);
+    try (Session session = SESSION_FACTORY.openSession()) {
+      Transaction tx = session.beginTransaction();
+      String[] parts = fullName.split("\\.");
+      if (parts.length != 3) {
+        throw new BaseException(
+            ErrorCode.INVALID_ARGUMENT, "Invalid registered model name: " + fullName);
+      }
+      String catalogName = parts[0];
+      String schemaName = parts[1];
+      String registeredModelName = parts[2];
+      try {
+        UUID schemaId = getSchemaId(session, catalogName, schemaName);
+        deleteRegisteredModel(session, schemaId, registeredModelName, force);
+        tx.commit();
+      } catch (RuntimeException e) {
+        if (tx != null && tx.getStatus().canRollback()) {
+          tx.rollback();
+        }
+        throw e;
+      }
+    }
+  }
+
+  public void deleteRegisteredModel(
+      Session session, UUID schemaId, String registeredModelName, boolean force) {
+    if (force) {
+      // Remove all model versions
```

Comment:
Please mention TODO : here

Reply:
I have a follow on PR ready to go once this lands that adds this in.